tests/test_encoding/test_decision_tree_encoder.py

```diff
@@ -2,24 +2,97 @@
 import pandas as pd
 import pytest
 
+from sklearn.exceptions import NotFittedError
+
 from feature_engine.encoding import DecisionTreeEncoder
 
 
-def test_encoding_method_param(df_enc):
-    # defaults
+# init parameters
+@pytest.mark.parametrize("enc_method", ["count", False, 1])
+def test_error_if_encoding_method_not_permitted_value(enc_method):
+    msg = (
+        "`encoding_method` takes only values 'ordered' and 'arbitrary'."
+        f" Got {enc_method} instead."
+    )
+    with pytest.raises(ValueError) as record:
+        DecisionTreeEncoder(encoding_method=enc_method)
+    assert str(record.value) == msg
+
+
+@pytest.mark.parametrize(
+    "unseen", ["string", False, ("raise", "ignore"), ["ignore"], np.nan]
+)
+def test_error_if_unseen_gets_not_permitted_value(unseen):
+    msg = (
+        'Parameter `unseen` takes only values "ignore", "raise", "encode". '
+        f"Got {unseen} instead."
+    )
+    with pytest.raises(ValueError) as record:
+        DecisionTreeEncoder(unseen=unseen)
+    str(record.value) == msg
+
+
+def test_error_if_unseen_is_encode_and_fill_value_is_none():
+    msg = (
+        "When `unseen='encode'` you need to pass a number to `fill_value`. "
+        f"Got {None} instead."
+    )
+    with pytest.raises(ValueError) as record:
+        DecisionTreeEncoder(unseen="encode", fill_value=None)
+    str(record.value) == msg
+
+
+@pytest.mark.parametrize("precision", ["string", 0.1, -1, np.nan])
+def test_error_if_precision_gets_not_permitted_value(precision):
+    msg = "Parameter `precision` takes integers or None. " f"Got {precision} instead."
+    with pytest.raises(ValueError) as record:
+        DecisionTreeEncoder(precision=precision)
+    str(record.value) == msg
+
+
+@pytest.mark.parametrize(
+    "params",
```

Comment:
Another IMO more readable way would be:

"encoding_method,ignore_format,precision,unseen,fill_value" instead of "params"
and then 
def test_init_param_assignment(encoding_method, ignore_format, precision, unseen, fill_value):
    DecisionTreeEncoder(
        encoding_method=encoding_method,
        ignore_format=ignore_format,
        precision=precision,
        unseen=unseen,
        fill_value=fill_value,
    )

Reply:
Done, thank you